### PR TITLE
Fix for MSYS2

### LINKED ISSUE #2: Doesn't work with pacman
I tried running paruz with just pacman and no other AUR helper and it fails to run. On exiting, the error is ```sudo pacman: no such command found```. I'm running:

- bash: 5.1.18
- fzf: 0.27.2

image

### PR DESCRIPTION
**Pull request by rashil2000**
MSYS2 uses pacman. This includes a tiny fix for it.

Also closes #2 

Tested on Arch Linux and MSYS2

### CODE REVIEW

File: paruz [hunk 1]
```diff
@@ -8,7 +8,7 @@ if [[ -z $PARUZ ]]; then
   elif ! command -v pacman >/dev/null 2>&1; then
     echo "Neither paru nor pacman found. Is this Arch?" >&2
     exit 1
-  elif [[ $EUID -eq 0 ]]; then
+  elif [[ $EUID -eq 0 ]] || [[ -f /usr/bin/msys-2.0.dll ]]; then
```
Comment (joehillen) [inline paruz:11]: You don't need to special case msys2. It's fixed by your next commit.

File: paruz [hunk 2]
```diff
@@ -78,12 +78,18 @@ if [[ ${#ARGS[@]} -eq 0 ]]; then
   ARGS=("-S")
 fi
 
+PRVW_CMD=''
+
+if [[ -f /usr/bin/msys-2.0.dll ]]; then
+  PRVW_CMD=/usr/bin/bash
```
Comment (joehillen) [inline paruz:84]: Why is this necessary? It can probably also be fixed by changing the shebang to `#!/usr/bin/env bash` which I forgot to do.

Comment (rashil2000) [reply paruz:84]: This is necessary because fzf on MSYS2 uses the `cmd` shell (as a default) for  executing the preview command. And `cmd` shell cannot understand shebangs. So we need to tell it the interpreter explicitly.

File: paruz (hunk 1 above)
Comment (rashil2000) [reply paruz:11]: There's no sudo on MSYS2. So we need to use just 'pacman'.

File: paruz (hunk 2 above)
Comment (joehillen) [reply paruz:84]: You can probably use `$0` instead. Try that and let me know if it works.

File: paruz (hunk 1 above)
Comment (joehillen) [reply paruz:11]: Are you able to set environment variables? If so, try setting `PARUZ=pacman`

File: paruz (hunk 2 above)
Comment (rashil2000) [reply paruz:84]: I tried. Doesn't work. 
The issue is, there's no concept of shebangs in `cmd`. As such, you can't execute script files as is (unless they have a known extension like .sh or .py) from the commandline. For such extensionless scripts, you need to specify the interpreter.

Comment (joehillen) [reply paruz:84]: I don't want to special case MSYS2. You could try linking `/usr/local/bin/paruz.sh` to `/usr/bin/paruz` or something.

File: paruz (hunk 1 above)
Comment (joehillen) [reply paruz:11]: I'm trying to fix your problem without a special case for MSYS2. If you `export PARUZ=pacman` in your shell, then it won't do the root check or try `sudo pacman`.

File: paruz (hunk 2 above)
Comment (rashil2000) [reply paruz:84]: Sorry, I must have missed it, but MSYS2 does not register itself as a runner of .sh files. So even if I run paruz.sh, it will fail unless I manually register a file association (something which most of the people do not do, and it's not recommended because of security risks.)
May I ask why you don't want a special case here? It doesn't affect any systems other than MSYS2.

Comment (rashil2000) [reply paruz:84]: I can see that the current implementation of checking the msys-2.0.dll file repeatedly can be cumbersome. I can make a variable for it so that we don't need to check it again (or use a cleaner method). Would that be alright?

File: paruz (hunk 1 above)
Comment (rashil2000) [reply paruz:11]: I'm again trying too understand what's the harm in having a special case (genuine question). If it's an implementation problem, see my comment below.

File: paruz (hunk 2 above)
Comment (joehillen) [reply paruz:84]: I think it's a better engineering practice to come up with a solution that works for everyone rather than add special cases. What about adding `/usr/bin/env bash`? That should work for everyone.

Comment (rashil2000) [reply paruz:84]: Yes, `/usr/bin/env bash` works. I'll make the change.

File: paruz (hunk 1 above)
Comment (rashil2000) [reply paruz:11]: Yes, it does.
But I find having to define PARUZ as 'pacman' slightly cumbersome, more so because there's already a fallback to 'pacman' in the script itself. 
There's already a condition running in the `if` block there. Would it be considered really bad if we just put another OR check there? (Sorry again, I'm just trying to understand. I get that PRVW_CMD was slightly invasive as a platform-specific fix, so I removed it as per your instructions. But isn't this one really insignificant in comparison?)

Comment (joehillen) [reply paruz:11]: Adding an environment variable shouldn't be cumbersome. I have like 100 environment variables in my shell config. This scenario is exactly why I added the variable.

Comment (rashil2000) [reply paruz:11]: Paruz was made to handle paru and pacman (as I understand from the script). I would have happily set the environment variable if I was using some other helper (as the Readme tells us). 

I will remove this condition if this feels really unnecessary.
